contracts/Vault.vy

```diff
@@ -1100,8 +1115,10 @@ def withdraw(
                 continue  # Nothing to withdraw from this Strategy, try the next one
 
             # Force withdraw amount from each Strategy in the order set by governance
+            preBalance: uint256 = self.token.balanceOf(self)
             loss: uint256 = Strategy(strategy).withdraw(amountNeeded)
```

Comment:
is it worth changing the withdraw API in BaseStrategy to return `amountFreed` value as well so we don't have to sandwich this function with balance checks?

https://github.com/yearn/yearn-vaults/blob/v0.4.3/contracts/BaseStrategy.sol#L801-L802

Reply:
That would be a great addition, but i think its out of scope since it breaks backward compatibility, all current strategies wont be able to be ported as is from 0.4.x to this release